presto-docs/src/main/sphinx/release/release-0.233.rst

```diff
@@ -0,0 +1,79 @@
+* Fix an issue where invalid checksum queries can be generated for certain queries containing columns of ``RowType``.
+* Fix an issue where checksum query would fail for queries containing map columns whose key or value types are arrays or rows.
+* Fix incorrect decision for determinism analysis of queries with top-level ``LIMIT`` clause. (:pr:`14176`).
+* Add checks for keys, values, and cardinality sum when validating a map column.
+* Add support to disable individual failure resolvers (:pr:`14148`).
+* Add support to auto-resolve control checksum query failures with ``COMPILER_ERROR``, instead of skipping the verification.
+* Add support for specifying non-deterministic catalogs by configuration property ``determinism.non-determinism-catalogs``.
+  Queries explicitly referencing tables from those catalogs are treated as non-deterministic.
+* Improve query performance during determinism analysis of queries with top-level ``LIMIT`` clause.
+* Improve correctness check for floating point columns whose mean values of either the control query or the test query is closed to 0.
+
+
+Druid Changes
+_____________
+* Add Druid Connector.
+* Improve query performance with filter pushdown.
+
+
+Geospatial Changes
+__________________
+* Improve :func:`ST_Points` to add support for major well-known spatial objects.
+  :func:`ST_Points` now supports ``POINT``, ``LINESTRING``, ``POLYGON``, ``MULTIPOINT``, ``MULTILINESTRING``, ``MULTIPOLYGON`` and ``GEOMETRYCOLLECTION``.
+* Improve :func:`ST_IsValid` and :func:`ST_IsSimple` to adhere to the ISO/OGC standards more closely.
+  The two functions used to return the same result but may now be different. Users should check both functions to be sure their geometries are well-behaved.
+  :func:`geometry_invalid_reason` will return different but semantically similar strings.
+* Improve performance of :func:`ST_Intersection` by reduces the cost by returning simply the geometry if it is with an enclosing envelope.
+  This can reduce the cost by ``10^5x`` or more for complex polygons when this occurs.
+
+SPI Changes
+___________
+* Add parameter ``NodeSelectionStrategy nodeSelectionStrategy`` in method ``ConnectorBucketNodeMap#createBucketNodeMap`` to indicate
+  which affinity strategy to use when creating a bucket node map.
+* Add parameter ``List<Node> sortedNodes`` in method ``ConnectorNodePartitioningProvider#getBucketNodeMap`` to provide
+  a sorted list of nodes from which a connector can choose to perform affinity scheduling.
+* Add enum ``NodeSelectionStrategy``. ``NO_PREFERENCE`` indicates data is remotely accessible from workers,
+  ``HARD_AFFINITY`` to indicate data and workers are collocated, and ``SOFT_AFFINITY`` to indicate data is remotely accessible
+  but scheduler will make the best effort to fetch the same piece of data from the same worker.
+* Replace ``ConnectorSplit#isRemoteAccessible`` with ``isRemoteAccessible`` is true if and only if ``getNodeSelectionStrategy`` returns ``HARD_AFFINITY``.
```

Comment:
Hard to interpret this

Reply:
@kewang1024 Similar here. If things cannot be explained concisely, just include one release note item and refer to the PR with:
```
(:pr:`13966`)
```

Reply:
* Add parameter ``NodeSelectionStrategy nodeSelectionStrategy`` in method ``ConnectorBucketNodeMap#createBucketNodeMap`` to indicate
  which affinity strategy to use when creating a bucket node map (:pr:`14095`).
* Add parameter ``List<Node> sortedNodes`` in method ``ConnectorNodePartitioningProvider#getBucketNodeMap`` to provide
  a list of nodes from which a connector can choose to perform affinity scheduling (:pr:`14095`).
* Add enum ``NodeSelectionStrategy``. ``NO_PREFERENCE`` indicates data is remotely accessible from workers,
  ``HARD_AFFINITY`` to indicate data and workers are collocated, and ``SOFT_AFFINITY`` to indicate data is remotely accessible (:pr:`13966`).
* Replace ``ConnectorSplit#isRemoteAccessible`` with ``isRemoteAccessible`` is true if and only if ``getNodeSelectionStrategy`` returns ``HARD_AFFINITY`` (:pr:`13966`).
* Replace ``ConnectorSplit#getAddresses`` with ``getPreferredNodes``, to provide hints to the scheduler where to schedule splits (:pr:`13966`).